server/jobs/legal_hold_job.go

```diff
@@ -148,6 +148,10 @@ func (j *LegalHoldJob) nextWaitInterval(now time.Time, metaData cluster.JobMetad
 	return delta
 }
 
+func (j *LegalHoldJob) RunFromAPI() {
+	j.processAllLegalHolds()
+}
+
 func (j *LegalHoldJob) run() {
 	j.client.Log.Info("Running Legal Hold Job")
 	exitSignal := make(chan struct{})
```

Comment:
It's unclear to me if the first sections of `run()` need to be run in the context of running from the API, or if they only apply in the production timed job context. I decided to extract the logic around processing the legal holds into its own function, and call that from the API handler.